### ISSUE TITLE
Webhook-triggers (templated pipelines) stop working on newer Spinnaker versions (1.10.1+)

### ISSUE THREAD

**Issue by iniinikoski** [spinnaker/spinnaker#3800]
### Issue Summary:

When Spinnaker's upgraded to latest version (1.11.2) (templated) pipelines with a webhook trigger stop working. The last working Spinnaker version running is 1.10.1, so atm we're blocked to upgrade.

### Cloud Provider(s):

N/A

### Environment:

Spinnaker running on top of GKE, deploying to both GKE and AWS.

### Feature Area: 

Pipeline triggering (Echo/Orca)

### Description:

Webhook triggers in templated pipelines should work after the upgrade.

### Steps to Reproduce:

I'm able to reproduce the behavior by just upgrading Spinnaker to version 1.11.2 and retriggering the webhook pipelines (which are templated). But @jtk54 has not been able to reproduce this in his test setup...

Also, this is the first feedback I've heard about this...

### Additional Details:

When the pipeline has been triggered, this one is logged in Orca's log:
`2019-01-03 07:58:31.731  INFO 1 --- [0.0-8083-exec-6] c.n.s.o.c.OperationsController           : [] Not starting pipeline (plan: true): 97576961-d68b-41b9-9ab4-c80981b8596`

However, if I plan the pipeline + template with roer - I get no errors.

I've also tried to get more info by changing log levels to debug earlier - so far it's been no use but I'll try to do that again soon.

---

**Comment by iniinikoski** [comment]
Tried also with the brand new stable, 1.11.3 - but no use - the issue still persists.

**Comment by ezimanyi** [comment]
I've just cherry-picked the fix for this issue to the 1.11 branch; you can test it out using `release-1.11.x-latest-unvalidated` or wait for the release of 1.11.4 which should be coming shortly.

**Comment by iniinikoski** [comment]
Upgraded Spinnaker to 1.11.4 - but I still get the same log line from Orca:

`2019-01-07 21:00:01.925  INFO 1 --- [0.0-8083-exec-7] c.n.s.o.c.OperationsController           : [] Not starting pipeline (plan: true): 06494dc5-e2bd-487c-9e39-226843f45d6c`

**Comment by ezimanyi** [comment]
@iniinikoski : That line is expected; when executing a templated pipeline, first a request is sent to Orca to convert the template into a fully hydrated pipeline (the request with `plan: true` you see in the logs), then a second request is sent to actually execute the pipeline.

The bug that was fixed in 1.11.4 was that this `plan` request was failing due to an unmatched expected artifact (which would have shown up as a failure message below that log line).  Do you see any other requests or errors below the above-quoted line?  If the pipeline isn't starting there may be another error below.

**Comment by iniinikoski** [comment]
Ah ok, thanks... Nope, no errors there. I'll try to raise the log level to see if I could get something out...

**Comment by iniinikoski** [comment]
@ezimanyi we've upgraded to the latest now - 1.12.2! :) I'll document the happenings here if someone else stumbles into this...

- The first issue what I had before was logging: when I sent the webhook-trigger - I was not able to see anything in logs - not even IMO in debug-level. Looks like a better logging was added - as I was able to get relevant information into orca's logs (I think it was orca here) immediately when I tried to do the trigger. So - a big thanks who had made improvements here! :)
- The second issue - which surfaced when I could get the proper error into the logs - was that we were resetting the payloadConstrains in the template - and Spinnaker did not like that at all. I dropped that from the template(s) -> a third issue surfaced :)
- Back then we were not using the Artifacts-support everywhere - as I had had some issues with it. This applied to our older templates. When changing the triggering + templates to use artifacts for delivering the needed information (and not sending it empty) - the issue was solved.

So, back to the originating:
- old Spinnaker version was 1.10.1
- the upgrade was done to 1.12.2 directly (tried multiple versions in between, from 1.10.X, 1.11.X without any results)
